packages/openid4vc/src/openid4vc-verifier/OpenId4VcSiopVerifierService.ts

```diff
@@ -0,0 +1,364 @@
+          definition: response.oid4vpSubmission.presentationDefinitions[0]?.definition,
+          presentations: response.oid4vpSubmission?.presentations.map(getVerifiablePresentationFromSphereonWrapped),
+        }
+      : undefined
+
+    const idToken = response.authorizationResponse.idToken
+      ? {
+          payload: await response.authorizationResponse.idToken.payload(),
+        }
+      : undefined
+
+    // TODO: do we need to verify whether idToken or vpToken is present?
+    // Or is that properly handled by sphereon's library?
+    return {
+      // Parameters related to ID Token.
+      idToken,
+
+      // Parameters related to DIF Presentation Exchange
+      presentationExchange,
+    }
+  }
+
+  public async getAllVerifiers(agentContext: AgentContext) {
+    return this.openId4VcVerifierRepository.getAll(agentContext)
+  }
+
+  public async getByVerifierId(agentContext: AgentContext, verifierId: string) {
+    return this.openId4VcVerifierRepository.getByVerifierId(agentContext, verifierId)
+  }
+
+  public async updateVerifier(agentContext: AgentContext, verifier: OpenId4VcVerifierRecord) {
+    return this.openId4VcVerifierRepository.update(agentContext, verifier)
+  }
+
+  public async createVerifier(agentContext: AgentContext) {
+    const openId4VcVerifier = new OpenId4VcVerifierRecord({
+      verifierId: utils.uuid(),
+    })
+
+    await this.openId4VcVerifierRepository.save(agentContext, openId4VcVerifier)
+    await storeActorIdForContextCorrelationId(agentContext, openId4VcVerifier.verifierId)
+    return openId4VcVerifier
+  }
+
+  private async getRelyingParty(
+    agentContext: AgentContext,
+    verifier: OpenId4VcVerifierRecord,
+    {
+      presentationDefinition,
+      requestSigner,
+      clientId,
+    }: {
+      presentationDefinition?: DifPresentationExchangeDefinition
+      requestSigner?: OpenId4VcJwtIssuer
+      clientId?: string
+    }
+  ) {
+    const authorizationResponseUrl = joinUriParts(this.config.baseUrl, [
+      verifier.verifierId,
+      this.config.authorizationEndpoint.endpointPath,
+    ])
+
+    const signatureSuiteRegistry = agentContext.dependencyManager.resolve(SignatureSuiteRegistry)
+
+    const supportedAlgs = getSupportedJwaSignatureAlgorithms(agentContext) as string[]
+    const supportedProofTypes = signatureSuiteRegistry.supportedProofTypes
+
+    // Check: audience must be set to the issuer with dynamic disc otherwise self-issued.me/v2.
+    const builder = RP.builder()
+
+    let _clientId = clientId
+    if (requestSigner) {
+      const suppliedSignature = await getSphereonSuppliedSignatureFromJwtIssuer(agentContext, requestSigner)
+      builder.withSignature(suppliedSignature)
+
+      _clientId = suppliedSignature.did
+    }
+
+    if (!_clientId) {
+      throw new AriesFrameworkError("Either 'requestSigner' or 'clientId' must be provided.")
+    }
+
+    // FIXME: we now manually remove did:peer, we should probably allow the user to configure this
```

Comment:
`did:peer` could still be valid right? Haven't seen it in any other context other than didcomm, but if you have exchanged it before hand it'll work. 

It can be left like this, as the user should likely define it themselves and by default just allow for all supported resolver methods.